theano/configdefaults.py

```diff
@@ -1524,11 +1536,24 @@ def _timeout_default():
 
 try:
     p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    cxx_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    cxx_version_str = 'CXX_NOT_FOUND'
+
+try:
+    p_out = output_subprocess_Popen(['g++', '-dumpversion'])
     gcc_version_str = p_out[0].strip().decode()
 except OSError:
     # Typically means gcc cannot be found.
-    gcc_version_str = 'GCC_NOT_FOUND'
+    gcc_version_str = 'CXX_NOT_FOUND'
 
+try:
+    p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    icc_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    icc_version_str = 'CXX_NOT_FOUND'
```

Comment:
Hi, @abergeron 
This seems a little crazy. 
ICC is a commercial compiler. It is a lot of different flag compared with gcc. And ICC support some features that gcc can't support good enough. Such as, vectorization, new cpu instruction and so on. And ICC also can't support arm and other CPU. These make the difference between ICC and g++ don't like the difference between g++ and clang++. And future, we will add some special optimized code to theano. These code maybe work well with gcc and other compiler. We need some more check and flag to make sure these code can work well with all compiler.
Thank you.